applications/GeoMechanicsApplication/geo_mechanics_application.cpp

```diff
@@ -0,0 +1,320 @@
+#include "geometries/triangle_3d_6.h"
+#include "geometries/quadrilateral_3d_4.h"
+#include "geometries/quadrilateral_3d_8.h"
+#include "geometries/quadrilateral_3d_9.h"
+#include "geometries/quadrilateral_interface_3d_4.h"
+#include "includes/variables.h"
+
+
+// Application includes
+#include "geo_mechanics_application.h"
+#include "geo_mechanics_application_variables.h"
+
+
+namespace Kratos {
+// We define the node type
+typedef Node<3> NodeType;
+
+KratosGeoMechanicsApplication::KratosGeoMechanicsApplication():
+    KratosApplication("GeoMechanicsApplication")
+
+    // small strain elements
+
+    // drained small strain elements
+
+    // undrained small strain elements
+
+    // small strain FIC elements
+
+    // small strain different order elements
+
+    // small strain interface elements
+
+    // Updated-Lagrangian elements
+
+    // Updated-Lagrangian different order elements
+
+    // geo structural elements
+
+    // conditions
+
+    {}
+
+void KratosGeoMechanicsApplication::Register() {
+    // calling base class register to register Kratos components
+    KratosApplication::Register();
+    KRATOS_INFO("") << " KRATOS___                             \n"
+                    << "     //   ) )                          \n"
+                    << "    //         ___      ___            \n"
+                    << "   //  ____  //___) ) //   ) )         \n"
+                    << "  //    / / //       //   / /          \n"
+                    << " ((____/ / ((____   ((___/ /  MECHANICS\n"
+                    << " Initializing KratosGeoMechanicsApplication..." << std::endl;
+
+
+
+    //Register Elements
+    // Small strain elements
+
+    // Drained small strain elements
+
+    // Undrained small strain elements
+
+    // Small strain FIC elements
+
+    // Small strain different order elements
+
+    // Small strain interface elements
+
+    // Updated-Lagranian elements
+
+    // Updated-Lagrangian different order elements
+
+    // Register geo structural elements
+
+    //Register Conditions
+
+    //Register Constitutive Laws
+
+    //Register Variables
+    KRATOS_REGISTER_VARIABLE( VELOCITY_COEFFICIENT )
+    KRATOS_REGISTER_VARIABLE( DT_PRESSURE_COEFFICIENT )
+
+    KRATOS_REGISTER_VARIABLE( DT_WATER_PRESSURE )
+    KRATOS_REGISTER_VARIABLE( NORMAL_FLUID_FLUX )
+
+    KRATOS_REGISTER_VARIABLE( DENSITY_SOLID )
+    KRATOS_REGISTER_VARIABLE( BULK_MODULUS_SOLID )
+    KRATOS_REGISTER_VARIABLE( BULK_MODULUS_FLUID )
+
+    KRATOS_REGISTER_VARIABLE( K0_MAIN_DIRECTION )
+    KRATOS_REGISTER_VARIABLE( K0_VALUE_XX )
+    KRATOS_REGISTER_VARIABLE( K0_VALUE_YY )
+    KRATOS_REGISTER_VARIABLE( K0_VALUE_ZZ )
+
+    KRATOS_REGISTER_VARIABLE( PERMEABILITY_XX )
+    KRATOS_REGISTER_VARIABLE( PERMEABILITY_YY )
+    KRATOS_REGISTER_VARIABLE( PERMEABILITY_ZZ )
+    KRATOS_REGISTER_VARIABLE( PERMEABILITY_XY )
+    KRATOS_REGISTER_VARIABLE( PERMEABILITY_YZ )
+    KRATOS_REGISTER_VARIABLE( PERMEABILITY_ZX )
```

Comment:
we have macros for tensor variables

Reply:
@loumalouomega can u point to one example of definition of those matrix variables and components?